onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -621,7 +624,7 @@ def _infer_Compress(self, node):
                                           output_shape))
 
     def _infer_Concat(self, node):
-        if any([i in self.sympy_data_ for i in node.input]):
+        if any([i in self.sympy_data_ or i in self.initializers_ for i in node.input]):
```

Comment:
This was an independent issue I found when I wrote the test. I can make a separate PR if necessary.